### PR TITLE
remove mamba's pin

### PR DESCRIPTION
**Pull request by ocefpaf**
Follow up on #102.

### CODE REVIEW

File: ioos_qc/qartod.py [hunk 1]
```diff
@@ -738,8 +738,9 @@ def attenuated_signal_test(inp : Sequence[N],
         def window_func(w):
             # When pandas>=1.0 and numba are installed, this is about twice as fast
             try:
+                import numba
                 return w.apply(np.ptp, raw=True, engine='numba')
-            except (ImportError, TypeError):
+            except (ImportError, TypeError, numba.core.errors.NumbaTypeError):
```
Comment (ocefpaf) [inline ioos_qc/qartod.py:743]: In a way numba never worked with masked arrays. See https://github.com/numba/numba/commit/25f17bbe24f99e1fb83468d0cf47b4b584472759

However, if we do want numba's speed-ups, we could try to use it when the Window inputs do not create NaNs and require masked-arrays. I could not figure this out yet so, for now, I'm proposing we just check for this numba error and let it default to non-numba behavior.  If/when we implement a masked-array workaround we should keep this check anyway to future proof any mask-array that may pass here.

**Comment by ocefpaf** [comment]
@iwensu0313 can you take a look at this one? it will allow us to install ioos_qc in environments  modern Python + mamba. Although, the mamba acceleration there won't be available. See my comment above on the reason why.

File: ioos_qc/qartod.py (hunk 1 above)
Comment (ocefpaf) [reply ioos_qc/qartod.py]: OK. I think I got to the bottom of this one. In fact, if the Windows produces NaN, then there is nothing we can do but skip numba and this implementation is OK to do so. However, if the Windows doesn't create NaN, numba will be used without a problem.

TL;DR numba acceleration will happen when the window parameters allow it.

**Comment by ocefpaf** [comment]
Self-merging this one b/c this will allow us to install `ioos_qc` in environments with modern numba.
